pagefaultgames/pokerogue
# [Move] Finish Alluring Voice, Burning Jealousy and Lash Out

## What are the changes the user will see?
Fully implements Alluring Voice, Burning Jealousy, and Lash Out.

## Why am I making these changes?
Reduce unimplemented moves.

## What are the changes from a developer perspective?
* Added `battleStatsIncrease: boolean` and `battleStatsDecrease: boolean` to `PokemonTurnData`
* Added `ConfuseIfBoostedAttr` class for Alluring Voice, which will confuse the target if their stats have been boosted this turn.
* Added `BurnIfBoostedAttr` class for Burning Jealousy, which will burn the target(s) if their stats have been boosted this turn.

### Screenshots/Videos

https://github.com/user-attachments/assets/b240fcbb-b95c-4811-bec2-8a1059f46406

## How to test the changes?
`npm run test alluring_voice`, `npm run test burning_jealousy`, `npm run test lash_out`
For Alluring Voice and Burning Jealousy, the following overrides should suffice for testing:
```ts
const overrides = {
  OPP_LEVEL_OVERRIDE: 10,
  OPP_SPECIES_OVERRIDE: Species.REGIELEKI,
  OPP_MOVESET_OVERRIDE: Array(4).fill(Moves.HOWL),
  OPP_ABILITY_OVERRIDE: Abilities.ICE_SCALES,
  STARTER_SPECIES_OVERRIDE: Species.SHUCKLE,
  MOVESET_OVERRIDE: [Moves.ALLURING_VOICE, Moves.BURNING_JEALOUSY],
  ABILITY_OVERRIDE: Abilities.BALL_FETCH,
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
Make sure to use Burning Jealousy before Alluring Voice so the enemy can't hit itself in confusion.
For Lash Out:
```ts
const overrides = {
  OPP_LEVEL_OVERRIDE: 10,
  OPP_SPECIES_OVERRIDE: Species.REGIELEKI,
  OPP_MOVESET_OVERRIDE: Array(4).fill(Moves.HOWL),
  OPP_ABILITY_OVERRIDE: Abilities.FUR_COAT,
  STARTER_SPECIES_OVERRIDE: Species.SHUCKLE,
  MOVESET_OVERRIDE: [Moves.LASH_OUT],
  ABILITY_OVERRIDE: Abilities.BALL_FETCH,
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
You'll need to watch the console output to verify the power is increased.

## Checklist
  - This is an updated version of #1619
